great_expectations/zep/interfaces.py

```diff
@@ -22,10 +41,10 @@ class Datasource(metaclass=MetaDatasource):
     # instance attrs
     name: str
     execution_engine: ExecutionEngine
-    assets: Dict[str, DataAsset]
+    assets: Mapping[str, DataAsset]
```

Comment:
I've changed this to a `Mapping` since I want subclasses to specify specific types of `DataAsset` subclasses. That is, the postgres datasource overwrites this as `assets: Mapping[str, TableAsset]`. `Dicts` are invariant types so don't allow this subclassing of the values.

Reply:
No typing errors here? 
Originally I had this defined as a Mapping but the Mapping interface doesn't include __setitem__ so it complained when I tried to set the assets.

Hence is my initial PR had a MutableMapping.

If this works, there's no issue.